gossip/src/cluster_info.rs

```diff
@@ -2183,15 +2184,23 @@ impl ClusterInfo {
         let _st = ScopedTimer::from(&self.stats.gossip_listen_loop_time);
         const RECV_TIMEOUT: Duration = Duration::from_secs(1);
         const SUBMIT_GOSSIP_STATS_INTERVAL: Duration = Duration::from_secs(2);
-        let mut packets = VecDeque::from(receiver.recv_timeout(RECV_TIMEOUT)?);
-        for payload in receiver.try_iter() {
-            packets.extend(payload);
-            let excess_count = packets.len().saturating_sub(MAX_GOSSIP_TRAFFIC);
-            if excess_count > 0 {
-                packets.drain(0..excess_count);
+        let mut num_packets = 0;
+        let mut packets = VecDeque::with_capacity(2);
+        for pkts in receiver
+            .recv_timeout(RECV_TIMEOUT)
+            .map(std::iter::once)?
+            .chain(receiver.try_iter())
+        {
+            num_packets += pkts.len();
+            packets.push_back(pkts);
+            while num_packets > MAX_GOSSIP_TRAFFIC {
+                let Some(num) = packets.pop_front().as_ref().map(Vec::len) else {
+                    break;
+                };
                 self.stats
                     .gossip_packets_dropped_count
-                    .add_relaxed(excess_count as u64);
+                    .add_relaxed(num as u64);
+                num_packets -= num;
```

Comment:
this chunk supersedes https://github.com/anza-xyz/agave/pull/4797?

Reply:
I don't understand what that means.

This commit is keeping the pipeline and load-shedding logic exactly the same as the old master.